c/xen_common.c

```diff
@@ -885,22 +885,27 @@ static int count_children(xmlNode *n, const char *name)
 static void destring(xen_session *s, xmlChar *name, const abstract_type *type,
                      void *value)
 {
-    switch (type->typename)
```

Comment:
I wanted to add the logging bit in line 906 and the compiler wouldn't accept the buffer declaration after the switch label.